src/x25519.rs

```diff
@@ -27,7 +105,7 @@ use rand_core::CryptoRng;
 /// # Returns
 ///
 /// A `Scalar`.
-fn decode_scalar(scalar: &[u8; 32]) -> Scalar {
+fn clamp_scalar(scalar: &[u8; 32]) -> Scalar {
```

Comment:
If `clamp_scalar` takes a `&Scalar` instead of a byte array, then the DH function above can be simplified.  The doc comment should also be updated to get rid of the references to "decode", and instead say something like "Clamps the scalar according to the X25519 specification".  Originally the API just had byte-oriented functions, so it made sense for it to take a byte array, but now it could have a more precise type.